### PR TITLE
Use `@guardian/tsconfig`

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Start using the `@guardian/tsconfig`’s recommended compiler options.

However, this disables `noUncheckedIndexedAccess`, as it would otherwise introduce many refactors and fixes. See #5777 for further discussions.

## Why?

Consistency accross client-side code styles.

### CODE REVIEW

File: dotcom-rendering/tsconfig.json [hunk 1]
```diff
@@ -1,72 +1,20 @@
...
     "baseUrl": "./",
-    /* Base directory to resolve non-absolute module names. */
-    /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
+    "forceConsistentCasingInFileNames": true,
+    "jsx": "react-jsx",
+    "jsxImportSource": "@emotion/react",
+    "lib": ["esnext", "es2015", "dom"],
+    "noEmit": true,
+    "noFallthroughCasesInSwitch": true,
+    "noUncheckedIndexedAccess": false, // @TODO: review this
```
Comment (oliverlloyd) [inline dotcom-rendering/tsconfig.json:12]: Not saying we shouldn't review this but I find these sorts of todos written in the code tend to hang around and then cause confusion \ disappointment later on. Maybe we replace it with a ticket that looks at each of these overrides?

**Review by oliverlloyd — APPROVED** [review]
This looks like a small change but I know a lot of work went into this!

File: dotcom-rendering/tsconfig.json (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/tsconfig.json:12]: Fair point, and there’s already a ticket!

```suggestion
    "noUncheckedIndexedAccess": false,
```

File: dotcom-rendering/tsconfig.json [hunk 2]
```diff
@@ -1,72 +1,20 @@
...
-    "moduleResolution": "node",
-    /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
+    "allowJs": true,
     "baseUrl": "./",
-    /* Base directory to resolve non-absolute module names. */
-    /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
+    "forceConsistentCasingInFileNames": true,
+    "jsx": "react-jsx",
+    "jsxImportSource": "@emotion/react",
+    "lib": ["esnext", "es2015", "dom"],
```
Comment (mxdvl) [inline dotcom-rendering/tsconfig.json:9]: I am wondering if `es2015` is redundant, there…

```suggestion
    "lib": ["esnext", "dom"],
```

To assess in a subsequent PR.

**Pull request merged**